iroha/src/account.rs

```diff
@@ -145,38 +145,44 @@ pub mod isi {
         }
     }
 
-    impl Transfer<Account, Asset, Account> {
+    impl Transfer<Account, Component, Account> {
         fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
-            world_state_view
+            let existing = world_state_view
                 .account(&self.source_id)
                 .ok_or("Failed to find accounts.")?
-                .assets
-                .get_mut(&self.object.id)
-                .expect("Asset not found.")
-                .quantity -= self.object.quantity;
+                .components
+                .get_mut(&self.object.id())
+                .expect("Asset's component was not found.");
+            if *existing < self.object {
```

Comment:
Also here I think the naming are difficult to understand. Comparing an `object` seems strange. Though I understand from where it comes, someone new to the code, might find it confusing.

Reply:
It's private API, so I think it's ok to have some inner generic names here, but code may be made better I guess.